frontend/src/design/components/UserModal.js

```diff
@@ -0,0 +1,102 @@
+import React, { useState, useEffect } from 'react';
+import {
+  Typography,
+  Button,
+  Box,
+  Table,
+  TableHead,
+  TableRow,
+  TableCell,
+  TableBody,
+  Dialog,
+  CircularProgress
+} from '@mui/material';
+import { useClient } from '../../services';
+import { useDispatch } from '../../globalErrors';
+import { listUserForGroup } from '../../services/graphql/Groups/listUserForGroup';
+
+const UserModal = ({ teams, open, onClose }) => {
+  const [loading, setLoading] = useState(false);
+  const [teamUsers, setTeamUsers] = useState([]);
+  const dispatch = useDispatch();
+  const client = useClient();
+
+  useEffect(() => {
+    const fetchUsers = async () => {
+      try {
+        setLoading(true);
+        const response = await client.query(listUserForGroup(teams)); // Use the GraphQL query function to fetch users in Team
+        if (response && response.data && response.data.listUsersForGroup) {
+          setTeamUsers(response.data.listUsersForGroup);
+        }
+      } catch (error) {
+      } finally {
+        setLoading(false);
+      }
+    };
+    if (client && teams) {
+      fetchUsers();
+    }
+  }, [dispatch, client, teams]);
+
+  return (
+    <Dialog maxWidth="lg" onClose={onClose} open={open}>
+      <Box
+        sx={{
+          padding: 10,
+          borderRadius: 5,
+          minWidth: 800,
+          maxHeight: 800,
+          overflowY: 'auto'
+        }}
+      >
+        <Typography id="team-members-title" variant="h6" color="textPrimary">
+          Team Members
+        </Typography>
+        {loading ? (
+          <Box
+            display="flex"
+            justifyContent="center"
+            alignItems="center"
+            minHeight="200px"
+          >
+            <CircularProgress />
+          </Box>
+        ) : (
+          <Table>
+            <TableHead>
+              <TableRow>
+                <TableCell>User ID</TableCell>
+              </TableRow>
+            </TableHead>
+            <TableBody>
+              {teamUsers.map((userId, index) => (
+                <TableRow key={index}>
+                  <TableCell>
+                    <Typography color="textSecondary" variant="body2">
+                      {userId}
+                    </Typography>
+                  </TableCell>
+                </TableRow>
+              ))}
+            </TableBody>
+          </Table>
```

Comment:
Is there a need to add a `<Pager>` element here and paginate the response returned from the backend (++ would likely have to add some sort condition on backend to ensure consistent results)

Otherwise what if a group have 100+ users?

Reply:
We don't currently have pagination support in our custom authorizer to match this.  In practice we don't have much more than about 20 users in any given team at the moment--the modal dialog will also just scroll through the list of users and the api calls are pretty quick.  Maybe an enhancement for the future?